docs/developer-guide/maps-configuration.md

```diff
@@ -669,6 +670,80 @@ PDOK.luchtfotoRGB
 PDOK.luchtfotoIR
 ```
 
+#### Vector
+
+The layer type vector is the type used for imported data (geojson, shapefile) or for annotations. Generally speaking, any vector data added directly to the map.
+This is the typical fields of a vector layer
+
+```json
+{
+    "type":"vector",
+    "features":[
+        {
+            "type":"Feature",
+            "geometry":{
+                "type":"Point",
+                "coordinates":[
+                12.516431808471681,
+                41.89817370656741
+                ]
+            },
+            "properties":{
+            },
+            "id":0
+        }
+    ],
+    "style":{
+        "weight":5,
+        "radius":10,
+        "opacity":1,
+        "fillOpacity":0.1,
+        "color":"rgba(0, 0, 255, 1)",
+        "fillColor":"rgba(0, 0, 255, 0.1)"
+    },
+    "hideLoading":true
+}
+```
+
+- `features`: features in GeoJSON format.
+- `style`: the style object.
+- `styleName`: name of a style to use (e.g. "marker").
+- `hideLoading`: boolean. if true, the loading will not be taken into account.
+
+#### Vector Style
+
+The `style` or `styleName` properties of vector layers (wfs, vector...) allow to apply a style the local data on the map.
+
+- `style`: a style object/array. It can have different format. In the simplest case thus is an object that uses some leaflet-like style properties:
+  - `weight`: width in pixel of the border / line.
+  - `radius`: radius of the circle (valid only for Point types)
+  - `opacity`: opacity of the border / line.
+  - `color`: color of the border / line.
+  - `fillOpacity`: opacity of the fill if any. (Polygons, Point)
+  - `fillColor`: color of the fill, if any. (Polygons, Point)
+- `styleName`: if set to `marker`, the `style` object will be ignored and it will use the default marker.
+
+In case of `vector` layer, style can be added also to the specific features. Other ways of defining the style for a vector layer have to be documented.
+
+#### WFS Layer
+
+A vector layer, and it adds to the map contents coming from WFS service. As the configuration point of view is somehow a mix between WMS and vector layer. it contains the search entry that allows to browse the data on the server side. The styling system is the same of the vector layer.
+
+This layer differs from the "vector" because all the loading/filtering/querying operations are applied directly using the WFS service, without storing anything locally.
+
+```json
+{
+    "type":"wfs",
+    "search":{
+        "url":"https://myserver.org/geoserver/wfs",
```

Comment:
I did this because it was the faster way to enable all WFS features originally though for WMS. 
There are many points where the presence of WFS (filters, featuregrid, charts, tables) use this field to identify the WFS presence. 
Do you think I should use a different solution ?

Reply:
Maybe we can just create a searchUrlSelector in all those places and code the logic there:

| layer.type === 'wfs' -> url
| otherwise -> search.rul

Reply:
![image](https://user-images.githubusercontent.com/1279510/81059141-30b91880-8ed0-11ea-8cb7-d4d2ada45c6c.png)

It's a little sparse here and there. Sometimes the whole layer is not passed I noticed, so not a quick fix. Maybe we can include this in a more general simplification and rationalization of configuration.